step-02/js/main.js

```diff
@@ -1,239 +1,278 @@
-callButton.disabled = true;
-hangupButton.disabled = true;
-startButton.onclick = start;
-callButton.onclick = call;
-hangupButton.onclick = hangup;
-
-var startTime;
-var localVideo = document.getElementById('localVideo');
-var remoteVideo = document.getElementById('remoteVideo');
-
-localVideo.addEventListener('loadedmetadata', function() {
-  trace('Local video videoWidth: ' + this.videoWidth +
-    'px,  videoHeight: ' + this.videoHeight + 'px');
-});
-
-remoteVideo.addEventListener('loadedmetadata', function() {
-  trace('Remote video videoWidth: ' + this.videoWidth +
-    'px,  videoHeight: ' + this.videoHeight + 'px');
-});
-
-remoteVideo.onresize = function() {
-  trace('Remote video size changed to ' +
-    remoteVideo.videoWidth + 'x' + remoteVideo.videoHeight);
-  // We'll use the first onresize callback as an indication that video has started
-  // playing out.
-  if (startTime) {
-    var elapsedTime = window.performance.now() - startTime;
-    trace('Setup time: ' + elapsedTime.toFixed(3) + 'ms');
+// Set up media stream constant and parameters.
+
+// On this codelab, you will be streaming video only (video: true).
+const mediaStreamConstraints = {
+  video: true,
+};
+
+// Set up to exchange only video.
+const offerOptions = {
+  offerToReceiveVideo: 1,
+};
+
+// Defines initial start time of the call (defined as connection between peers).
+let startTime = null;
+
+// Define peer connections, streams and video elements.
+const localVideo = document.getElementById('localVideo');
+const remoteVideo = document.getElementById('remoteVideo');
+
+let localStream;
+let remoteStream;
+
+let localPeerConnection;
+let remotePeerConnection;
+
+
+// Define MediaStreams callbacks.
+
+// Handles success by adding the MediaStream to the video element.
+const gotLocalMediaStream = (mediaStream) => {
+  localVideo.srcObject = mediaStream;
+  localStream = mediaStream;
+  trace('Received local stream.');
+  callButton.disabled = false;  // Enable call button.
+};
+
+// Handles error by logging a message to the console with the error message.
+const localMediaStreamError = (error) => {
+  trace(`navigator.getUserMedia error: ${error.toString()}.`);
+};
+
+// Handles remote MediaSteam success by...
+const gotRemoteMediaStream = (event) => {
+  const mediaStream = event.stream;
+  remoteVideo.srcObject = mediaStream;
+  remoteStream = mediaStream;
+  trace('Remote peer connection received remote stream.');
+};
+
+
+// Add behavior for video streams.
+
+// Logs a message with the id and size of a video element.
+const logVideoLoaded = (event) => {
+  const video = event.target;
+  trace(`${video.id} videoWidth: ${video.videoWidth}px, ` +
+        `videoHeight: ${video.videoHeight}px.`);
+};
+
+// Logs a message with the id and size of a video element.
+const logResizedVideo = (event) => {
+  logVideoLoaded(event);  // Log size of video by calling logVideoLoaded.
+
+  // First on-resize is used as an indication that video started playing out.
```

Comment:
I think this should be a comment on the function itself, something like this:

'This function is added as a handler for the video resize event. This event is fired when video begins streaming.'

Reply:
Done.